src/sendNotify.ts

```diff
@@ -20,13 +19,15 @@ export const sendDiscordNotification = async (
       },
       body: JSON.stringify(requestBody),
     });
-    if (response.ok) {
-      return { success: true };
-    } else {
-      return { success: false, message: `Discord通知送信失敗, HTTP status: ${response.status}` };
+    if (!response.ok) {
+      throw new Error({
+        message: `Discord通知送信失敗, HTTP status: ${response.status}`,
+      });
     }
   } catch (error) {
     console.error('Discord通知送信失敗:', error);
-    return { success: false, message: (error as Error).message };
+    throw new Error({
+      message: (error as Error).message,
+    });
```

Comment:
CIで型エラーになっている件は、以下のようにすればとりあえず直ると思います！
（Errorを投げるようにするのかどうかという判断の話は別で必要）

```suggestion
    throw new Error((error as Error).message);
```

そして、さらに踏み込むと、Errorコンストラクタの第2引数は `cause` を受け取れるようになっていて、ここに元のエラーをセットしておくと、エラーの連鎖を表現できて、何か起きたときに分かりやすいかも

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/Error#cause

```suggestion
    throw new Error((error as Error).message, { cause: error });
```